recipes/fftw/all/conanfile.py

```diff
@@ -70,64 +87,88 @@ def generate(self):
         tc.variables["ENABLE_OPENMP"] = self.options.openmp
         tc.variables["ENABLE_THREADS"] = self.options.threads
         tc.variables["WITH_COMBINED_THREADS"] = self.options.get_safe("combinedthreads", False)
-        tc.variables["ENABLE_FLOAT"] = self.options.precision == "single"
-        tc.variables["ENABLE_LONG_DOUBLE"] = self.options.precision == "longdouble"
         tc.variables["ENABLE_SSE"] = self.options.simd == "sse"
         tc.variables["ENABLE_SSE2"] = self.options.simd == "sse2"
         tc.variables["ENABLE_AVX"] = self.options.simd == "avx"
         tc.variables["ENABLE_AVX2"] = self.options.simd == "avx2"
         tc.generate()
 
+    @property
+    def build_folder(self):
+        bf = super().build_folder
+        return os.path.join(bf, self._current_precision) if self._current_precision else bf
+
+    @property
+    def _all_precisions(self):
+        return [p for p in ALL if self.options.get_safe(f"precision_{p}")]
+
     def build(self):
+        def on_off(value):
+            return "ON" if value else 'OFF'
+
         apply_conandata_patches(self)
-        cmake = CMake(self)
-        cmake.configure()
-        cmake.build()
+        for self._current_precision in self._all_precisions:
+            cmake = CMake(self)
+            variables = {
+                "ENABLE_FLOAT": on_off(self._current_precision == SINGLE),
+                "ENABLE_LONG_DOUBLE": on_off(self._current_precision == LONGDOUBLE),
+                "EENABLE_QUAD_PRECISION": on_off(self._current_precision == QUAD)
+            }
+            cmake.configure(variables=variables)
+            cmake.build()
 
     def package(self):
         copy(self, "COPYRIGHT", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
-        cmake = CMake(self)
-        cmake.install()
-        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
-        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        for self._current_precision in self._all_precisions:
+            cmake = CMake(self)
+            cmake.install()
+            rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        prec_suffix = self._prec_suffix[str(self.options.precision)]
-        cmake_config_name = "FFTW3" + prec_suffix
-        cmake_namespace = "FFTW3"
-        cmake_target_name = "fftw3" + prec_suffix
-        pkgconfig_name = "fftw3" + prec_suffix
-        lib_name = "fftw3" + prec_suffix
+        cmake_config_name = cmake_namespace = "FFTW3"
```

Comment:
It's a regression here, there should be a FIXME at least. Each lib has a different CMake config file, and it can't be modeled by conan with this change in fftw.

Reply:
Why not just create one recipe per fftw flavor?

Reply:
@SpaceIm Thank you for your thoughts! Indeed we discussed about the a proxy recipe option with separated recipes, but we opted for a single recipe and a next PR creating those separated CMake config files. You see, other package managers follow the same approach, is a bit messy, but should work.

Reply:
> and a next PR creating those separated CMake config files

How? Do you plan to address https://github.com/conan-io/conan/issues/9000?

Reply:
@SpaceIm That feature will require more time to be done. The PR in question is a workaround for this recipe.

Reply:
Like exposing a custom cmake config file which would include expected config file generated by cmakedeps?

Reply:
That's it!

Reply:
@uilianries can we add more examples to the docs on how to properly model CMake properties?

for example, while working on this recipe, I was unable to find any guidance on how to generate config files with different file names per targets.